### ISSUE TITLE
Global executables are preferred over bin_dir

### ISSUE THREAD

**Issue by jyggen** [phpro/grumphp#140]
Greetings! 

We're running into some weird issues because GrumPHP currently prefers globally installed executables located in `PATH` over project-specific ones located in `bin_dir`.

For example we have a dependency on PHPUnit 4.8 in our `composer.json` (which thus is installed to `./vendor/bin`, configured as `bin_dir` in `grumphp.yml`). However when running GrumPHP we get an error stating the following:

```console
PHP Fatal error:  Call to undefined method PHPUnit_TextUI_TestRunner::printVersionString() in /usr/share/php/PHPUnit/TextUI/Command.php on line 816
```

(it actually took some fiddling around to get GrumPHP to output this, but I'll create a separate issue for that)

In this case the developer have an old version (3.7.28) of PHPUnit installed which GrumPHP prefers over what the project actually uses. I've also seen this happen when installing 

The issue seems to be that `GrumPHP\Locator\ExternalCommand::locate()` sends in `bin_dir` as an extra directory to `Symfony\Component\Process\ExecutableFinder::find()`. `find()` actually appends the extra directories to the end, meaning that it'll first go through each directory defined in `PATH` before looking in `bin_dir`. 

This is how the extra directory (`bin_dir` in GrumPHP's case) is handled by the Symfony class (Line 69-72 in `ExecutableFinder.php`):

```php
$dirs = array_merge(
    explode(PATH_SEPARATOR, getenv('PATH') ?: getenv('Path')),
    $extraDirs
);
```

Is this the indented behaviour or just something no one's been bothered by yet? :)

**Comment by veewee** [comment]
I was also bothered by this in the past.
Since it is possible to install grumphp globally, it is hard to determine where the external commands are located.
The ExternalCommand locator was added to make it possible to use both locally installed packages and globally installed packages. It also handles compatibility with windows, where the command could be named `phpunit.bat` for example. So you could say this is intended behaviour.

Personally I just try to keep all my globally installed packages up to date. This is how I prevent the problem you are facing. I can see that this could be a problem in the scenario you are describing.

There is not really a clean solution for this other than writing our own ExecutableFinder or overwriting both environment variables and open_basedirs.

Do you have any suggestions on how we could improve this?

**Comment by veewee** [comment]
This one is fixed in #148.
